src/v/cluster/topic_recovery_validator.cc

```diff
@@ -0,0 +1,376 @@
+    // allow enough quota just to check existance and self metadata check
+    // and few segments existance
+    // TODO ensure that future evolution of anomalies detector operate on a
+    // "metadata first" behaviour
+    auto detector = cloud_storage::anomalies_detector{
+      std::move(bucket),
+      ntp,
+      rev_id,
+      remote,
+      rtc_logger,
+      rtc.root_abort_source()};
+
+    // spin up detector
+    auto anomalies_fut = co_await ss::coroutine::as_future(detector.run(
+      rtc,
+      cloud_storage::anomalies_detector::segment_depth_t{
+        checks.max_segment_depth}));
+
+    if (anomalies_fut.failed()) {
+        // propagate shutdown exceptions, but treat other exceptions as hard
+        // failures. likely could be serde exceptions for a corrupt file
+        auto ex = anomalies_fut.get_exception();
+        if (ssx::is_shutdown_exception(ex)) {
+            co_return ss::coroutine::exception(ex);
+        }
+        vlog(
+          clusterlog.error,
+          "{}/{} ({}) manifest metadata check threw exception: {}",
+          ntp,
+          rev_id,
+          get_path(),
+          ex);
+        co_return failure;
+    }
+
+    // examine anomalies, some are real failures while other can be
+    // tollerated
+    auto anomalies = anomalies_fut.get();
+    if (anomalies.status == cloud_storage::scrub_status::failed) {
+        // download issue. could be a configuration issue or a s3 issue
+        vlog(
+          clusterlog.error,
+          "{}/{} ({}) manifest metadata check download failure",
+          ntp,
+          rev_id,
+          get_path());
+        co_return download_issue;
+    }
+
+    if (!anomalies.detected.has_value()) {
+        // no anomalies detected, everything is ok up to segment depth
+        vlog(clusterlog.info, "{}/{} manifest metadata check ok", ntp, rev_id);
+        co_return passed;
+    }
+
+    // this unpack is to trigger a compilation error if a new field is added
+    // to cloud_storage::anomalies. If a new anomalies is added, it should
+    // be also handled here
+    auto const& [missing_partition_manifest, missing_spillover_manifests, missing_segments, segment_anomalies, ignored_1, ignored_2, ignored_3, ignored_4]
+      = anomalies.detected;
+
+    if (missing_partition_manifest) {
+        // missing manifest anomaly can be ok: this means the partition did
+        // not have time to upload data
+        vlog(
+          clusterlog.info,
+          "{}/{} manifest metadata check: missing manifest",
+          ntp,
+          rev_id);
+        co_return missing_manifest;
+    }
+
+    if (!missing_segments.empty()) {
+        // missing segments could be problematic and need an operator fixing
+        // the manifest before proceeding
+        vlog(
+          clusterlog.error,
+          "{}/{} ({}) manifest metadata: missing segments",
+          ntp,
+          rev_id,
+          get_path());
+        co_return failure;
```

Comment:
nit: maybe it's worth renaming to `anomaly_detected` or something? That way it's less ambiguous how this is different from `download_issue`

Reply:
good point